### PR TITLE
Create more fact groups for ease of use. Closes #233.

### LINKED ISSUE #233: Support rhel, jboss, and all fact groups
## Specify type:
- Enhancement
___

## Description:
Create some more “fact groups” we currently have “default” which is all facts and is now the implicit choice. We could have “rhel”, “jboss”, and “all” fact groups where “all” is the new implicit choice. "rhel"  is everything minus the jboss.x facts and “jboss” is the jboss.x facts.

### PR DESCRIPTION
**Pull request by chambridge**

### CODE REVIEW

File: rho/scancommand.py [hunk 1]
```diff
@@ -399,15 +399,21 @@ def _validate_options(self):
 
         # perform fact validation
         facts = self.options.facts
-        if facts == [] or facts == ['default']:
-            self.facts_to_collect = list(utilities.DEFAULT_FACTS_TUPLE)
+        if facts == [] or facts == ['default'] or facts == ['all']:
+            self.facts_to_collect = list(utilities.DEFAULT_FACTS)
+        elif facts == ['jboss']:
+            self.facts_to_collect = list(utilities.JBOSS_FACTS +
```
Comment (elyezer) [inline rho/scancommand.py:405]: What is the difference between a fact group that has the `_TUPLE` suffix and the ones that does not have it?

**Review by elyezer — APPROVED** [review]
The code looks good

File: rho/scancommand.py (hunk 1 above)
Comment (chambridge) [reply rho/scancommand.py:405]: `JBOSS_FACTS_TUPLE = ('jboss.installed-versions','jboss.deploy-dates', 'jboss.running-versions')`

`JBOSS_FACTS = JBOSS_FACTS_TUPLE + BRMS_FACTS_TUPLE + FUSE_FACTS_TUPLE`

Comment (elyezer) [reply rho/scancommand.py:405]: Thanks

**Pull request merged**